cmake_modules/PaRSECConfig.cmake.in

```diff
@@ -69,6 +69,7 @@ endif(@PARSEC_HAVE_CUDA@)
 
 if(@PARSEC_HAVE_HIP@)
   enable_language(CXX)
+  list(APPEND CMAKE_SYSTEM_PREFIX_PATH /opt/rocm)
```

Comment:
Can we get the path for the ROCM we detected instead of hard-coding it?

Reply:
Yes, this was just an attempt to get some kind of reasonable default in place so that the HIP version of parsec would at least be usable sometimes.  Right now, I think it will always fail to use the parsec if HIP was enabled because it can't find HIP.

I will look into how to get the "real" location of HIP that the parsec was built against.

Reply:
After looking at the current parsec source, I'm even more unsure of the correct approach.  The top-level CMakeLists.txt is using a hard-coded "/opt/rocm" for the HIP search, so my original fix was consistent with that.  However, it seems like the PaRSECConfig.cmake shouldn't even be doing a find_package(HIP) in the first place.  Shouldn't it just reflect the original results of find_package(HIP) from the initial parsec build?  If you do a new find_package() when something wants to use PaRSECConfig.cmake, then it is possible for it to find a different HIP than parsec was built with.

Reply:
The "/opt/rocm" is coming from how the FindHIP works, i.e. unlike FindCUDA which allows for a root dir, it will take the root dir from `CMAKE_SYSTEM_PREFIX_PATH`. Awful !!!

PaRSECConfig.cmake is used by other projects that to pull parsec as an external module. It is supposed to contain everything detected by the installed parsec, allowing the other project to have the same configuration as the installed parsec. As such, we need to add the same value to `CMAKE_SYSTEM_PREFIX_PATH` and `CUDA_TOOLKIT_ROOT_DIR`, and in general the same configuration for everything including hwloc and MPI.

Reply:
Yes I agree that PaRSECConfig.cmake should provide all the same build parameters to projects that want to know that information.  However, it seems like running find_package() in the PaRSECConfig.cmake is the wrong way to do this.  Shouldn't it just return same the original results from find_package() (from the parsec build) and return those?  Doesn't running find_package() again risk finding something different?

Reply:
That would have been the best solution assuming all FindPackage would set the same variables, and that only these variables will be ever used. Unfortunately, that's not the case, and reexecuting the FindPackage under the same conditions as in the parsec package is the most sensible approach.

Reply:
OK then, should is add back the `list(APPEND CMAKE_SYSTEM_PREFIX_PATH /opt/rocm)` to the Config.cmake, 

Or should I implement something better both in PaRSECConfig.cmake and in the top-level CMakeLists.txt?

In HeFFT, the following is done, which seems like an improvement:

```
    if (ROCM_PATH)
        list (APPEND CMAKE_PREFIX_PATH ${ROCM_PATH}/hip ${ROCM_PATH})
    elseif (DEFINED ENV{ROCM_PATH})
        list (APPEND CMAKE_PREFIX_PATH ENV{ROCM_PATH}/hip ENV{ROCM_PATH})
    elseif (IS_DIRECTORY /opt/rocm)
        list (APPEND CMAKE_PREFIX_PATH /opt/rocm/hip /opt/rocm)
    endif()
    enable_language(HIP)
    find_package(rocfft REQUIRED)
```

Reply:
Nice! To be consistent with other FindPackage please use ROCM_ROOT_DIR instead of ROCM_PATH.

Reply:
OK, I've updated the CMake so that the parameter ROCM_ROOT_DIR is available and defaults to /opt/rocm if it is not set and /opt/rocm exists.  Then in PaRSECConfig.cmake, it takes this value for the CMAKE_PREFIX_PATH. In my tests, this works both if you set it with `-DROCM_ROOT_DIR=something` and if you let it use the default.  Here is a CI job where dplasma is building against a parsec install with this new cmake config: https://github.com/ICLDisco/parsec/actions/runs/16758340841/job/47446780770